vspec/model/vsstree.py

```diff
@@ -73,6 +73,9 @@ def __init__(self, name, source_dict: dict, parent=None, children=None, break_on
         self.type = VSSType.from_str(source_dict["type"])
         self.uuid = source_dict["uuid"]
 
+        if "$file_name$" not in source_dict.keys():
+            source_dict["$file_name$"]="UNKNOWN"
+
```

Comment:
In my opinion, `$file_name$` should use `source_dict["$file_name$"]=None` and then either guarantee to have `$file_name$` or use something along these lines:
```python
if "$file_name$" in source_dict:
    raise FileNameNotFoundException
```

Or else if this continues like this, later it could show errors like  `Warning: UNKNOWN filename does not follow the vss naming pattern`

Reply:
In fact i believe to use $file_name$ in the source_dict is a bad idea in general. It's better to just send as a mandatory argument in the constructor.

Reply:
let it crash->That may be useful (instead of "hiding" some error).

 Removing it from source_dict: I do not think so, _if_ I understand the code correctly the source dict is the (preprocessed) YAML, that is is extended with additional metadata during earlier parsing steps, so it really helps traceability to have such information "directly" near the data, should problem arise, and not splitting it out and "manually" taking care of this association.

Reply:
Removed the check. Now it will crash and burn, if $file_name$ is not there instead of hiding the error.